src/watchNPM.js

```diff
@@ -0,0 +1,44 @@
+const request = require('request-promise')
+const _ = require('lodash')
+const ChangesStream = require('changes-stream')
+const clean = require('normalize-registry-metadata')
+const logger = require('~/src/logging').logger(module)
+
+// will hit up registry
+exports.checkWatcher = async function () { // TODO: fill the skeleton
+  console.log('todo')
+}
+
+// will publish to mq
+exports.handleChanges = async function ({channel, client, registry}) {
+  logger.info('saw change....todo handle logic')
+}
+
+exports.startNPMWatcher = async function ({channel, client, registry}) {
+  console.log(registry)
+  client.get('changes-stream', function (err, reply) {
+    if (err) {
+      logger.error('error when hitting redis')
+      throw Error(err)
+    }
+    logger.info('Reply from Redis: ' + JSON.stringify(reply))
+    logger.info('starting changes stream')
+
+    const changes = new ChangesStream({
+      db: registry,
+      since: 'now',
+      include_docs: true
+    })
+
+    changes.on('error', function (error) {
+      changes.destroy()
+      logger.error('error in changesStream')
+      throw Error(error)
+    })
+
+    changes.on('data', function () {
+      exports.handleChanges({channel, client, registry})
+    })
+    return changes
+  })
+}
```

Comment:
I would probably just scrap this file and move the `changes-stream` code you have here over to whatever file you plan on handling your 'loop' (mentioned in my above comments). I don't think exporting separate functions to handle the small bits of the same loop would have made things any easier/cleaner.